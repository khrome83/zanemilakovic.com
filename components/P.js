// @ts-check --jsx

import PropTypes from 'prop-types';

const P = ({ children, className }) => (
  <p className={`p ${className}`}>
    { children }
    <style jsx>{`
      @import './styles/variables.css';

      .p {
        margin: 0;
        line-height: var(--line-height-min);
        padding-bottom: var(--line-height-min);
        max-width: var(--max-text-width);

        @media (--tablet-viewport) {
          line-height: var(--line-height-mid);
          padding-bottom: var(--line-height-mid);
        }

        @media (--desktop-viewport) {
          line-height: var(--line-height-max);
          padding-bottom:  var(--line-height-max);
        }
      }
    `}</style>
  </p>
);

P.defaultProps = {
  className: null,
};

P.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
};

export { P as default };
