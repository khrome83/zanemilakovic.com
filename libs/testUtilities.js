// @ts-check --jsx
/* eslint-disable no-param-reassign */

const cleanSnapshots = (node) => {
  for (const prop in node) {
    if (prop === 'className') {
      const className = node[prop].replace(/jsx-[0-9]+[\s]?/g, '');
      node.className = className || '';
      if (!node.className) {
        delete node.className;
      }
    } else if (typeof node[prop] === 'object') {
      cleanSnapshots(node[prop]);
    }
  }
};

const lib = {
  cleanSnapshots,
};

export { lib as default, cleanSnapshots };
