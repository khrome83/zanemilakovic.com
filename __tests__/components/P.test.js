import React from 'react';
import renderer from 'react-test-renderer';
import { cleanSnapshots } from '../../libs/testUtilities';
import P from '../../components/P';

test('P renders correctly', () => {
  const component = renderer.create(<P className="large-font">Test to Render</P>);
  const tree = component.toJSON();
  cleanSnapshots(tree);
  expect(tree).toMatchSnapshot();
});
