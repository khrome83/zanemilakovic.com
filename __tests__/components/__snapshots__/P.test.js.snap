// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`P renders correctly 1`] = `
<p
  className="p large-font"
>
  Test to Render
</p>
`;
